src/nemdeling/nemdeling.controller.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  Body,
+  Controller,
+  InternalServerErrorException,
+  Logger,
+  Post,
+  ServiceUnavailableException,
+} from "@nestjs/common";
+import { DisplayApiService } from "../display-api/display-api.service";
+import {
+  ApiBody,
+  ApiCreatedResponse,
+  ApiInternalServerErrorResponse,
+  ApiTags,
+} from "@nestjs/swagger";
+import { NemDelingService } from "./nemdeling.service";
+import { ServiceMessageBody, NemDelingResult, NemdelingSlide } from "./types";
+import { ApiServiceUnavailableResponse } from "@nestjs/swagger/dist/decorators/api-response.decorator";
+
+type RequestNormalizerResult = { [key: string]: NemdelingSlide[] };
+
+@ApiTags("NemDeling")
+@Controller("api/v1/nemdeling")
+export class NemDelingController {
+  constructor(
+    private readonly nemDelingService: NemDelingService,
+    private readonly displayApiService: DisplayApiService,
+    private readonly logger: Logger
+  ) {}
+
+  /**
+   * The template type to use when creating service message slides.
+   */
+  private readonly serviceMessageTemplateType = "Servicemeddelelse";
+
+  @Post("service-messages")
+  @ApiCreatedResponse({
+    description: "The service message was synced.",
+    type: String,
+  })
+  @ApiServiceUnavailableResponse({
+    description: "Application is busy processing.",
+  })
+  @ApiInternalServerErrorResponse({
+    description: "Application not able to process request.",
+  })
+  @ApiBody({ type: String })
+  async serviceMessage(@Body() body: ServiceMessageBody): Promise<string> {
+    const results: NemDelingResult[] = [];
+
+    if (
+      this.nemDelingService.serviceMessagesAreSyncing > 0 &&
+      this.nemDelingService.serviceMessagesAreSyncing < 5
+    ) {
+      this.nemDelingService.serviceMessagesAreSyncing++;
+      throw new ServiceUnavailableException("Service messages are already being synced.");
+    }
+
+    this.nemDelingService.serviceMessagesAreSyncing++;
+
+    const data = await this.serviceMessagesDataMapper(body);
+    for (const [screenName, slides] of Object.entries(data.result)) {
+      const playlist = await this.nemDelingService.getServiceMessagePlaylistFromScreenName(
+        screenName
+      );
+      if (!playlist) {
+        continue;
+      }
+
+      const result = await this.nemDelingService.syncPlaylist(playlist, slides);
+      results.push({
+        name: screenName,
+        status: result ? "success" : "error",
+      });
+    }
+
+    data.notFound.forEach((screenName) => {
+      results.push({
+        name: screenName,
+        status: "not_found",
+      });
+    });
+
+    this.nemDelingService.serviceMessagesAreSyncing = 0;
+    this.logger.log("Service messages result: " + JSON.stringify(results));
+    return "OK";
```

Comment:
If we have errors we may as well output that even though we still do 200?

Reply:
This is the repsonse to NemDeling whi doesn't care, but I guess it doesn't hurt